src/cargo/util/toml/embedded.rs

```diff
@@ -159,8 +205,74 @@ fn sanitize_name(name: &str) -> String {
     name
 }
 
+struct Source<'s> {
+    shebang: Option<&'s str>,
+    info: Option<&'s str>,
+    frontmatter: Option<&'s str>,
+    content: &'s str,
+}
+
+fn split_source(input: &str) -> CargoResult<Source<'_>> {
+    let mut source = Source {
+        shebang: None,
+        info: None,
+        frontmatter: None,
+        content: input,
+    };
+
+    if source.content.starts_with("#![") {
```

Comment:
I need to update this to match https://github.com/rust-lang/rust/blob/7d9bce327332fef22cf687b44de58f06d30daeb0/compiler/rustc_lexer/src/lib.rs#L229